modules/issue_tracker/jsx/WatchingCheckbox.js

```diff
@@ -0,0 +1,35 @@
+class WatchingCheckbox extends React.Component {
+  constructor(prop) {
+    super();
+    this.state = {
+      checked: prop.checked,
+      issueId: prop.issue_id
+    };
+    this.onChange = this.onChange.bind(this);
+  }
+  render() {
+    return (
+      <td>
+        <input type = "checkbox" checked = {this.state.checked} onChange = {this.onChange}/>
+      </td>
+    );
+  }
+  onChange(e) {
+    var that = this;
```

Comment:
1. Use `const` or `let` instead of `var` in all ES6 supported classes (a.k.a in all files that run through webpack)
2. Instead of using `self` or `that`, just bind `this` to success callback. e.g:
```js
success: function(data) {
  this.setState({checked: data.watching});
}.bind(this)
```